larpandoracontent/LArMonitoring/TwoViewTransverseTracksValidationTool.cc

```diff
@@ -0,0 +1,164 @@
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+bool TwoViewTransverseTracksValidationTool::IsSameMCParticle(const Cluster *const pCluster1, const Cluster *const pCluster2)
+{
+    const MCParticle *const pMCParticle1(MCParticleHelper::GetMainMCParticle(pCluster1));
+    const MCParticle *const pMCParticle2(MCParticleHelper::GetMainMCParticle(pCluster2));
+
+    if (pMCParticle1 && pMCParticle2)
+    {
+        return pMCParticle1->GetUid() == pMCParticle2->GetUid();
+    }
+
+    return false;
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+void TwoViewTransverseTracksValidationTool::CollectTruthInformation(const Cluster *const pCluster, TreeDataBox &treeDataBox, std::string clusterName)
+{
+    constexpr float floatDefault(std::numeric_limits<int>::lowest());
+
+    int foundMCParticle(0);
+    int pdgMCParticle(std::numeric_limits<int>::lowest());
+    int isPrimaryMCParticle(0);
+    float energyMCParticle(floatDefault);
+    float momXMCParticle(floatDefault), momYMCParticle(floatDefault), momZMCParticle(floatDefault);
+    float posXMCParticle(floatDefault), posYMCParticle(floatDefault), posZMCParticle(floatDefault);
+
+    const MCParticle *const pMCParticle(MCParticleHelper::GetMainMCParticle(pCluster));
+    if (pMCParticle)
+    {
+        foundMCParticle = 1;
+        pdgMCParticle = pMCParticle->GetParticleId();
+        isPrimaryMCParticle = static_cast<int>(pMCParticle->IsRootParticle());
+        energyMCParticle = pMCParticle->GetEnergy();
+        momXMCParticle = pMCParticle->GetMomentum().GetX();
+        momYMCParticle = pMCParticle->GetMomentum().GetY();
+        momZMCParticle = pMCParticle->GetMomentum().GetZ();
+        posXMCParticle = pMCParticle->GetVertex().GetX();
+        posYMCParticle = pMCParticle->GetVertex().GetY();
+        posZMCParticle = pMCParticle->GetVertex().GetZ();
+    }
+
+    StoreAndRegisterDatum(foundMCParticle, ("foundMCParticle" + clusterName), treeDataBox);
+    StoreAndRegisterDatum(pdgMCParticle, ("pdgMCParticle" + clusterName), treeDataBox);
+    StoreAndRegisterDatum(isPrimaryMCParticle, ("isPrimaryMCParticle" + clusterName), treeDataBox);
+    StoreAndRegisterDatum(energyMCParticle, ("energyMCParticle" + clusterName), treeDataBox);
+    StoreAndRegisterDatum(momXMCParticle, ("momXMCParticle" + clusterName), treeDataBox);
+    StoreAndRegisterDatum(momYMCParticle, ("momYMCParticle" + clusterName), treeDataBox);
+    StoreAndRegisterDatum(momZMCParticle, ("momZMCParticle" + clusterName), treeDataBox);
+    StoreAndRegisterDatum(posXMCParticle, ("posXMCParticle" + clusterName), treeDataBox);
+    StoreAndRegisterDatum(posYMCParticle, ("posYMCParticle" + clusterName), treeDataBox);
+    StoreAndRegisterDatum(posZMCParticle, ("posZMCParticle" + clusterName), treeDataBox);
+
+    return;
```

Comment:
I mentioned this in a similar comment by Andy.
I really like functions to have a well defined return at the end (completely personal preference obviously!).
If this clashes with house style then I am happy to change but otherwise I'd rather leave in place...